### DISCUSSION TITLE
Can Whisper annotate named entities?

### DISCUSSION THREAD

**Discussion by mirix** [openai/whisper#1485] [Q&A]
Hello,

Does Whisper have any explicit or implicit understanding of NER? 

Should it be able to label named entities?

I am asking because one of the issues I am finding is that, if you split sentences on ". ", punctuated named entities such as "U.S. " or "John W. Doe"  pose a problem. 

Of course, this issue can be addressed a posteriori, but one could save considerable resources if we had them annotated a priori.

Best,

Ed

**Comment by glangford** [comment]
You cannot do this in Whisper, however you can post-process using a tool such as spaCy to identify sentence breaks, named entities etc. For example. it is possible to post-process a word-level timestamped .json to create an .srt which follows sentence boundaries and breaks lines at grammatically selected boundaries.

**Comment by mirix** [comment]
Thanks! My current approach:

https://github.com/mirix/approaches-to-diarisation
